### PR TITLE
fixes cacheadapter so that recyclerview gets populated

### PR DESCRIPTION
**Pull request by michelejoseph1**
### Summary:
_Context_ - This is part of a larger effort to create an Android app that takes in a list of ingredients and returns a list of cocktail recipes that can be made using them.

_Reference_ - [RapidAPI](https://rapidapi.com/thecocktaildb/api/the-cocktail-db) and [TheCocktailDB](https://www.thecocktaildb.com/api.php)

_This Diff_ - makes cacheadapter populate the recyclerview properly so that the recipe name, image, and description are pulled from the cache to save time and help with system optimization

### Test Plan:
When there are recipes containing the inserted ingredients: 
device-2022-07-11-122232 (1)

Where there are no recipes containing all of the inserted ingredients (returns empty recycler view): 
device-2022-07-11-124850

### CODE REVIEW

**Comment by michelejoseph1** [comment]
adjusting now so that null ingredients do not appear on screen!

File: app/src/main/java/com/example/cocktailme/MainActivity.java [hunk 1]
```diff
@@ -26,7 +26,7 @@ public class MainActivity extends AppCompatActivity {
     ImageButton settingsButton;
     Button generateButton;
     String insertedIngredients;
-    private DatabaseHelper db;
+    private DatabaseHelper databaseHelper;
```
Comment (alfonsocheng) [inline app/src/main/java/com/example/cocktailme/MainActivity.java:29]: great job renaming this! definitely helps!

File: app/src/main/java/com/example/cocktailme/CocktailNamesActivity.java [hunk 2]
```diff
@@ -65,5 +93,77 @@ public boolean onNavigationItemSelected(@NonNull MenuItem menuItem) {
...
+            }
+            @Override
+            public void onFailure(int statusCode, Headers headers, String response, Throwable throwable) {
+                Log.d(TAG, "onFailure" + statusCode + response);
+            }
+        });
     }
+
+
+        //goRecipeDetailsFromCache();
```
Comment (alfonsocheng) [inline app/src/main/java/com/example/cocktailme/CocktailNamesActivity.java:137]: we should remove the extra comments

File: app/src/main/java/com/example/cocktailme/CocktailNamesActivity.java [hunk 3]
```diff
@@ -65,5 +93,77 @@ public boolean onNavigationItemSelected(@NonNull MenuItem menuItem) {
...
+        getRecipesMethod(insertedIngredients);
+
+    }
+//doesn't work with commas
+    public void getRecipesMethod(String insertedIngredients) {
+        RequestHeaders headers = new RequestHeaders();
+        RequestParams params = new RequestParams();
+        params.put("i", insertedIngredients);
+        Log.d(TAG, "results: " + insertedIngredients);
+        headers.put("X-RapidAPI-Key", "c13dadbdfdmshb5f916990392087p1e49ccjsnae8ca04335f1");
```
Comment (alfonsocheng) [inline app/src/main/java/com/example/cocktailme/CocktailNamesActivity.java:109]: it probably makes sense to put these into they're own file or at least as constants

File: app/src/main/java/com/example/cocktailme/fragments/HomeFragment.java [hunk 4]
```diff
@@ -77,123 +74,63 @@ public static HomeFragment newInstance(int index) {
...
                 RecipeResponse recipeResponse = response.body();
-                verticalRecipeModels = recipeResponse.getRecipeModels();
-                if (verticalRecipeModels != null) {
-                    adapter = new CacheAdapter(verticalRecipeModels, HomeFragment.this);
-                    recyclerView.setAdapter(adapter);
+                listRecipeModels = recipeResponse.getRecipeModels();
+                if (listRecipeModels != null) {
+                    cacheAdapter = new CacheAdapter(listRecipeModels, HomeFragment.this);
+                    recyclerView.setAdapter(cacheAdapter);
                 }else {
```
Comment (alfonsocheng) [inline app/src/main/java/com/example/cocktailme/fragments/HomeFragment.java:119]: nit: space before else
`} else {`

File: app/src/main/java/com/example/cocktailme/fragments/HomeFragment.java [hunk 5]
```diff
@@ -77,123 +74,63 @@ public static HomeFragment newInstance(int index) {
...
-                    adapter = new CacheAdapter(verticalRecipeModels, HomeFragment.this);
-                    recyclerView.setAdapter(adapter);
-                    Toast.makeText(getActivity(), "No Result Found", Toast.LENGTH_LONG).show();
+                    cacheAdapter = new CacheAdapter(listRecipeModels, HomeFragment.this);
+                    recyclerView.setAdapter(cacheAdapter);
+                    Toast.makeText(getActivity(), "Unfortunately, we did not find any recipes containing those ingredients.", Toast.LENGTH_LONG).show();
```
Comment (alfonsocheng) [inline app/src/main/java/com/example/cocktailme/fragments/HomeFragment.java:122]: How come the toast doesn't show up at the bottom?

File: app/src/main/java/com/example/cocktailme/CocktailNamesActivity.java [hunk 6]
```diff
@@ -65,5 +93,77 @@ public boolean onNavigationItemSelected(@NonNull MenuItem menuItem) {
...
         });
         bottomNavigationView.setSelectedItemId(R.id.action_home);
+
+
+
+
+        getRecipesMethod(insertedIngredients);
+
+    }
+//doesn't work with commas
```
Comment (alfonsocheng) [inline app/src/main/java/com/example/cocktailme/CocktailNamesActivity.java:103]: not sure what this comment is referring to ?

File: app/src/main/java/com/example/cocktailme/db/RecipeModel.java [hunk 7]
```diff
@@ -212,7 +215,7 @@ public RecipeModel(Parcel in) {
 
     }
```
Comment (alfonsocheng) [inline app/src/main/java/com/example/cocktailme/db/RecipeModel.java:217]: I think it'll be useful here to add comments on why we're counting to 15

File: app/src/main/java/com/example/cocktailme/db/DatabaseHelper.java [hunk 8]
```diff
@@ -113,26 +113,42 @@ public void insertRecipe(RecipeModel recipeModel) {
...
-        values.put(RecipeModel.INGREDIENT12, ingredient12);
-        values.put(RecipeModel.INGREDIENT13, ingredient13);
-        values.put(RecipeModel.INGREDIENT14, ingredient14);
-        values.put(RecipeModel.INGREDIENT15, ingredient15);
-        values.put(RecipeModel.MEASUREMENT1, measurement1);
-        values.put(RecipeModel.MEASUREMENT2, measurement2);
-        values.put(RecipeModel.MEASUREMENT3, measurement3);
-        values.put(RecipeModel.MEASUREMENT4, measurement4);
-        values.put(RecipeModel.MEASUREMENT5, measurement5);
+
```
Comment (alfonsocheng) [inline app/src/main/java/com/example/cocktailme/db/DatabaseHelper.java:124]: I think the extra space was an accident, let's revert to what was there before

File: app/src/main/java/com/example/cocktailme/db/RecipeModel.java [hunk 9]
```diff
@@ -164,13 +168,12 @@ public class RecipeModel implements Parcelable {
...
-
-
     public static final String DELETE_TABLE =
             "DROP TABLE IF EXISTS " + TABLE_NAME;
 
-    public RecipeModel() {}
-
+    public RecipeModel() {
+    }
+//we are counting to 15 because there is a maximum of 15 possible ingredients/measurements for each recipe
```
Comment (alfonsocheng) [inline app/src/main/java/com/example/cocktailme/db/RecipeModel.java:176]: nit, comments have  space after slashes for readability; eg.
```
// here is a comment
```

File: app/src/main/java/com/example/cocktailme/fragments/HomeFragment.java [hunk 10]
```diff
@@ -214,11 +151,10 @@ public void onResponse(Call<RecipeResponse> call, Response<RecipeResponse> respo
                 }
                 if (nRandom < 20) {
```
Comment (alfonsocheng) [inline app/src/main/java/com/example/cocktailme/fragments/HomeFragment.java:152]: for my understanding, what's nRandom for, and what does is it doing?

**Review by alfonsocheng — APPROVED** [review]
lgtm 🚀

File: app/src/main/java/com/example/cocktailme/fragments/HomeFragment.java (hunk 10 above)
Comment (michelejoseph1) [reply app/src/main/java/com/example/cocktailme/fragments/HomeFragment.java:152]: nRandom is the position of the recipeModel in the list. it fetches the recipe from the cache and checks if retrofit is null and if it is, a blank GsonConverterFactory is created and if not, the recipe is populated it to the recyclerview

Comment (alfonsocheng) [reply app/src/main/java/com/example/cocktailme/fragments/HomeFragment.java:152]: So is it accurate to say that up to 20 recipes can be kept in the cache at one time?

**Pull request merged**